src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -33,27 +33,35 @@ public class Frame<TContext> : Frame
             {
                 while (!_requestProcessingStopping)
                 {
-                    while (!_requestProcessingStopping && !TakeStartLine(SocketInput))
+                    while (!_requestProcessingStopping && TakeStartLine(SocketInput) != RequestLineStatus.Done)
                     {
                         if (SocketInput.RemoteIntakeFin)
                         {
                             // We need to attempt to consume start lines and headers even after
                             // SocketInput.RemoteIntakeFin is set to true to ensure we don't close a
                             // connection without giving the application a chance to respond to a request
                             // sent immediately before the a FIN from the client.
-                            if (TakeStartLine(SocketInput))
+                            var requestLineStatus = TakeStartLine(SocketInput);
+                            if (requestLineStatus != RequestLineStatus.Empty &&
+                                requestLineStatus != RequestLineStatus.Done)
```

Comment:
It it's done parsing, you should break. If the request line is empty you should return.